runeq/config.py

```diff
@@ -38,9 +39,39 @@
 """
 
 
-class Config:
+class BaseConfig:
+    """Base class to hold configuration for accessing Rune APIs."""
+
+    graph_url: str
+    stream_url: str
+
+    def refresh_auth(self) -> bool:
+        """
+        Refresh authentication credentials if possible, returning a bool
+        indicating if the refresh occurred.
+
+        This is specific to the authentication style: e.g. it may
+        be implemented to refresh a JWT. By default, it is a no-op.
+
+        The API clients contain logic to catch possible authentication
+        errors, invoke this method, and retry the request (if credentials
+        are successfully refreshed).
+
+        """
+        return False
```

Comment:
Maybe this should return nothing and raise an exception (which will never happen in the no-op implementation) if something goes wrong, rather than returning a boolean that should sometimes be ignored? Not a blocker of course since this would change something (and really I guess nothing is stopping an implementation from raising an exception anyway...)

Reply:
This is existing functionality -- when an API request fails with a suspected auth error, the graph & stream clients use this boolean to decide whether to retry the request (if tokens were successfully refreshed) or raise the original exception